bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductteaserImpl.java

```diff
@@ -0,0 +1,147 @@
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.text.NumberFormat;
+import java.util.List;
+import java.util.Locale;
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = Productteaser.class, resourceType = ProductteaserImpl.RESOURCE_TYPE)
+public class ProductteaserImpl implements Productteaser {
+
+    protected static final String RESOURCE_TYPE = "venia/components/commerce/product/v1/Productteaser";
+    private static final Logger LOG = LoggerFactory.getLogger(ProductteaserImpl.class);
+    private static final String PRODUCT_PATH_PROP = "productSku";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @Inject
+    private Resource resource;
+
+    @Inject
+    private Page currentPage;
+
+    private ProductInterface product;
+
+
+    private NumberFormat priceFormatter;
+
+
+    @Inject
+    private ProductGraphqlService productGraphqlService;
+
+
+    @ScriptVariable
+    private ValueMap properties;
+
+    private Page productPage;
+
+    @PostConstruct
+    private void initModel() {
+        String sku = null;
+        String productPath = null;
+        productPage = Utils.getProductPage(currentPage);
+        if (productPage == null) {
+            productPage = currentPage;
+        }
+        productPath = properties.get(PRODUCT_PATH_PROP, String.class);
+        if (productPath != null && !productPath.isEmpty()) {
+            sku = getSkuFromUrl(productPath);
+            GraphqlClient client = resource.adaptTo(GraphqlClient.class);
+            if (client == null) {
+                LOG.error("Cannot get a GraphqlClient using the resource at {}", resource.getPath());
+            }
+            Locale locale = currentPage.getLanguage(false);
+            this.product = productGraphqlService.getProduct(client, sku);
+            this.priceFormatter = Utils.buildPriceFormatter(locale, this.getCurrency());
+        }
+    }
+
+    @Override
+    public String getName() {
+        return (this.product != null ? this.product.getName() : null);
+
+    }
+
+    @Override
+    public String getFormattedPrice() {
+
+        Double price = this.getPrice();
+
+        if (price != null) {
+            return this.priceFormatter.format(price);
+        }
+        return null;
+    }
+
+    @Override
+    public String getUrl() {
+        return (this.product != null ? String.format("%s.%s.html", productPage.getPath(), product.getUrlKey()) : null);
```

Comment:
Please do not duplicate that from `ProductListItemImpl`, better move this to the `Util` class and change the code here and in `ProductListItemImpl`.